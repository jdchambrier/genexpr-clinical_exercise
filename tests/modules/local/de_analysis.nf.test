nextflow_process {

    name "Test Process DE_ANALYSIS"
    script "modules/local/de_analysis.nf"
    process "DE_ANALYSIS"

    test("Should run without failures") {

        when {
            process {
                """
                input[0] = file("${projectDir}/test_data/clean_clinical.csv")
                input[1] = "MALE"
                input[2] = file("${projectDir}/bin/de_analysis.Rmd")
                container = 'rpreprocess:0.1'
                """
            }
        }

        then {
            assert process.success
            
            with(process.out){
                assert significance_results.size() == 1
                assert snapshot(significance_results).match("significance_results")
                assert d2_report.size() == 1
                def lines = file(d2_report[0]).readLines()
                assert snapshot(lines[0..10]).match("d2_report_10lines")
                assert versions.size() == 1
                assert snapshot(versions).match("versions")
            }
        }

    }

}
