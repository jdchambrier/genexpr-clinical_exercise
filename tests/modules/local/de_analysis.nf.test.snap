{
    "significance_results": {
        "content": [
            [
                "significance_results.csv:md5,010535c9595d6fd90b74ba830d4a7d66"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-07-27T14:29:51.947105"
    },
    "versions": {
        "content": [
            [
                "versions.yml:md5,21aec93b73ab25259433f6e2fc933d65"
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-07-27T14:29:51.978954"
    },
    "d2_report_10lines": {
        "content": [
            [
                "<!DOCTYPE html>",
                "",
                "<html>",
                "",
                "<head>",
                "",
                "<meta charset=\"utf-8\" />",
                "<meta name=\"generator\" content=\"pandoc\" />",
                "<meta http-equiv=\"X-UA-Compatible\" content=\"IE=EDGE\" />",
                "",
                ""
            ]
        ],
        "meta": {
            "nf-test": "0.9.2",
            "nextflow": "23.10.1"
        },
        "timestamp": "2025-07-27T14:29:51.975627"
    }
}